lib/rules/selector-max-universal/index.js

```diff
@@ -39,14 +52,28 @@ const rule = (primary) => {
 		 * @param {import('postcss').Rule} ruleNode
 		 */
 		function checkSelector(selectorNode, ruleNode) {
-			const count = selectorNode.reduce((total, childNode) => {
+			const count = selectorNode.reduce((total, childNode, index, nodes) => {
 				// Only traverse inside actual selectors
 				// All logical combinations will be resolved as nested selector in `postcss-resolve-nested-selector`
 				if (childNode.type === 'selector') {
 					checkSelector(childNode, ruleNode);
 				}
 
-				if (childNode.type === 'universal') total += 1;
+				let prevChildNode = nodes[index - 1];
+				let prevChildNodeValue = prevChildNode && prevChildNode.value;
+
+				// checks if previous child node is descendant combinator
+				if (prevChildNodeValue === ' ') {
+					prevChildNode = nodes[index - 2];
```

Comment:
[suggest]

- What about using the `.prev()` method? It can make `nodes` and `index` unnecessary.
- What about using `.type === 'combinator'` to check a descendant combinator? The code intention becomes clearer.

```suggestion
				let prevChildNode = childNode.prev();
				let prevChildNodeValue = prevChildNode && prevChildNode.value;

				// checks if previous child node is descendant combinator
				if (prevChildNode && prevChildNode.type === 'combinator' && prevChildNodeValue === ' ') {
					prevChildNode = prevChildNode.prev();
```